src/test/java/ru/spbstu/reader/VcfReaderTest.java

```diff
@@ -0,0 +1,63 @@
+package ru.spbstu.reader;
+
+import org.junit.jupiter.api.Test;
+import ru.spbstu.reader.dto.VcfRecord;
+
+import java.io.ByteArrayInputStream;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class VcfReaderTest {
+
+    @Test
+    public void testReadVcfFile() throws IOException {
+        List<VcfRecord> result = VcfReader.read(new FileInputStream("src/test/resources/test_vcf_file.vcf"));
+        assertEquals(8, result.size());
+    }
+
+    @Test
+    public void testReadVcfRecord() throws IOException {
+        final String headers = "##fileformat=VCFv4.0\n" +
+                "#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\tNA12878";
+        final String chrom = "chr1";
+        final int pos = 115;
+        final String id = "ss";
+        final String ref = "A";
+        final String alt = "T";
+        final Map<String, Object> info = Map.of(
+                "AC", "1",
+                "AF", "0.50",
+                "AN", "2",
+                "DP", "1019",
+                "MQ", "19.20"
+        );
+        final String text = String.format(
+                "%s\n%s\t%s\t%s\t%s\t%s\t0\tFDRtranche2.00to10.00+\t%s\tGT:AD:DP:GL:GQ"
+                        + "\t0/1:610,327:308:-316.30,-95.47,-803.03:99",
+                headers, chrom, pos, id, ref, alt, mapToString(info));
+
+        List<VcfRecord> result = VcfReader.read(new ByteArrayInputStream(text.getBytes()));
+
+        assertEquals(1, result.size());
+
+        VcfRecord record = result.get(0);
+        assertEquals(chrom, record.getChrom());
```

Comment:
Might better create expectedRecord and use one method assertEquals:
```
final VcfRecord expectedRecors = VcfRecord.builder()
                .chrom("chr1")
                .pos(115)
                .id("ss")
                .ref(Allele.REF_A)
                .alt(Allele.ALT_T)
                .info(info)
                .build();
 assertEquals(expectedRecors, record);
```

Reply:
It's better to use `assertThat` and `isEqualTo` methods. It's more human readble.

Reply:
I'm not sure why we're testing just one method, but It's ok, we wouldn't need read(final String path) and read(final VCFIterator iterator) for now.

If you extend you first test to validate records list, you don't need the second test.